test/unit/test_amqpclient.js

```diff
@@ -106,7 +106,7 @@ describe('AMQPClient', function() {
 
       return client.connect(mock_uri)
         .then(function () {
-          return client.createSender(queue, { name: 'queue_TX' });
+          return client.createSender(queue, { options: { name: 'queue_TX' } });
```

Comment:
Also, we could maintain the name as a top level option (the relevant change for this is in `u.linkName`), but that does sort of add a special case when I _think_ the goal here is to standardize everything.. 

Writing `client.createReceiver(address, { name: 'something' })` feels very natural to me, whereas `client.createReceiver(address, { attachOptions: { name: 'something' } });` feels very awkward.. *shrug*